SSP IM1 Revision

bmaclach opened the pull request:
IM1 and its derivation were updated to match the manual version. As part of this, a new assumption and two new data definitions were added, as they are used in the derivation.

smiths commented:
I've compared the generated version to the case study.  Looks great.  I do have a few minor comments
- x_wt is missing from the inputs in the Drasil code
- in the first equation for the derivation, should it be N'_i (should it be written as the effective force, with the prime symbol)
- I know it is planned for later, but we really should make FS be F_S, as it currently is written causes some confusion for reading the equations
- for DD: convertFunc1 and DD: convertFunc2, the labels in Drasil are Constant, they should be "First Function for ..." etc
- for DD: convertFunc1 and DD: convertFunc2, the subscripts do not seem to be correct.  The psi and the phi are missing subscripts, and for the second function there is a subscript n-1, instead of i-1

bmaclach commented:
Thanks @smiths. To address each item:

- This is an interesting case. I left x_wt out because it's actually not used in any of the equations (the x-coordinate is implicit when we write an equation for a value indexed by i). That being said, we probably should include it anyway since you don't know which y-coordinate to use without the corresponding x-coordinate. If you agree, I'll go ahead and add it.
- Yes, fixed in 24a194346be0358e282884bba9ea048266d195d7
- Okay, changed the symbol in 24a194346be0358e282884bba9ea048266d195d7
- Changed in 24a194346be0358e282884bba9ea048266d195d7
- The n-1 was a mistake, fixed in 24a194346be0358e282884bba9ea048266d195d7. As for the missing indices, there is a limitation where the left-hand side of a DD can't currently be an indexed symbol. This is discussed in #359.

smiths commented:
Sounds good @bmaclach.  Yes, I think we should put x_wt back in.  We could always remove it later, but for now I would like the Drasil version to match the manual case study version.  For now, I think we should still keep the two versions in sync.  However, we might be getting close to the time where we stop "maintaining" the manual version.

Thanks for making the updates.  I had forgotten about issue #359.  Thank you for the reminder.  😄
